lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3869,4 +3869,41 @@ function ( $carry, $item ) {
 		$theme_json->theme_json['styles'] = self::convert_variables_to_value( $styles, $vars );
 		return $theme_json;
 	}
+
+	/**
+	 * Generates a selector for a block style variation.
+	 *
+	 * @param string $variation_name Name of the block style variation.
+	 * @param string $block_selector CSS selector for the block.
+	 *
+	 * @return string Block selector with block style variation selector added to it.
+	 */
+	public static function get_block_style_variation_selector( $variation_name, $block_selector ) {
+		$variation_class = ".is-style-$variation_name";
+
+		if ( ! $block_selector ) {
+			return $variation_class;
+		}
+
+		$selector_parts = explode( ',', $block_selector );
+		$result         = array();
+
+		foreach ( $selector_parts as $part ) {
+			$class_added = false;
+			$result[]    = preg_replace_callback(
+				'/((?::is|not|has|where\([^)]+\))?\s*)([^\s:]+)/',
```

Comment:
I'm having a little trouble wrapping my head around this regex, I'm sorry! Why do we need to name the valid pseudo-classes to match against?

From looking at the tests, it sounds like we always want the variation class name to be attached to the outer wrapper / the very first part of the selector, directly preceding any `:` character or whitespace?

Also, if it's being updated in PHP here, would we need a corresponding change in [the JS equivalent](https://github.com/WordPress/gutenberg/blob/e103afb02dc3c814cb95f37a85bc5117cc8458a1/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L1074)?

Reply:
> I'm having a little trouble wrapping my head around this regex, I'm sorry! 

So do I and I wrote it! 😅 

> Why do we need to name the valid pseudo-classes to match against?

I was thinking it might be necessary to restrict the potential pseudo functions that get matched but this is probably overkill. Without them works, passes the tests, and is likely more flexible. I've cleaned up the regex in [478ed9e](https://github.com/WordPress/gutenberg/pull/58051/commits/478ed9e4bc3c6a52d305b23c1d95d695a70b4531), thanks!

> Also, if it's being updated in PHP here, would we need a corresponding change in [the JS equivalent](https://github.com/WordPress/gutenberg/blob/e103afb02dc3c814cb95f37a85bc5117cc8458a1/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L1074)?

Tunnel vision on my part. I often get caught out with the split between PHP and JS.